onnxruntime/core/session/abi_session_options.cc

```diff
@@ -144,9 +144,9 @@ ORT_API_STATUS_IMPL(OrtApis::SetIntraOpNumThreads, _Inout_ OrtSessionOptions* op
 #ifdef _OPENMP
   ORT_UNUSED_PARAMETER(options);
   ORT_UNUSED_PARAMETER(intra_op_num_threads);
-  LOGS_DEFAULT(WARNING) << "Since openmp is enabled in this build, this API cannot be used to configure"
-                           " intra op num threads. Please use the openmp environment variables to control"
-                           " the number of threads.";
+  std::cout << "WARNING: Since openmp is enabled in this build, this API cannot be used to configure"
+               " intra op num threads. Please use the openmp environment variables to control"
```

Comment:
Would it be better to check if there is a default logger and use that if possible? Would require an update to the LoggingManager to add the ability to check. Not a big deal for this particular use case but if we have more places where we need to support this sort of usage it would be nice to be as consistent as possible with where messages go. 

Separate question: are we fixing the right issue? The usage of LOGS_DEFAULT without a default logger should throw, but in the github issue the user said it failed silently. If we fixed that and returned a message about a default logger being needed would that have been good enough?

Reply:
For your first question - so far there's only one place where this is needed. I could check for the availability of the default logger, but I guess that would make it inconsistent since this warning might show up in the logger sink in one case (when env is setup before) and in the cout stream in another case (when env is not setup before).

For your second question - it does throw, however, for some reason it doesn't show up when using the C# APIs. In a C/C++ program, you will see an exception being thrown. Returning a message about a default logger being needed wouldn't help because you're not necessarily required to create an env before setting up session options. There's no dependency between the two. Moreover, in the C# API, there's no concept of env; you create the session options object first and then the session with the session options obj as an argument; the env is created in the session constr.

Reply:
OK. Do we need to give some more thought to the C# API? If we implicitly create the env with a session does that incorrectly create a 1:1 relationship between the two? Wondering if it might be better to make the separation between the env and a session a bit more explicit.

Reply:
A comment saying why we're using std::cout instead of the default logger would be helpful too.

Reply:
> OK. Do we need to give some more thought to the C# API? If we implicitly create the env with a session does that incorrectly create a 1:1 relationship between the two? Wondering if it might be better to make the separation between the env and a session a bit more explicit.

May be. Currently all our bindings follow a similar pattern in that the env is not exposed directly. I think the obvious reason was simplicity of using the APIs.